packages/builder/src/stores/builder/datasources.ts

```diff
@@ -300,6 +300,49 @@ export class DatasourceStore extends DerivedBudiStore<
     const info = await API.fetchInfoForDatasource(datasource)
     return info.tableNames || []
   }
+
+  async getViewNames(datasource: Datasource) {
+    try {
+      const info = await API.fetchViewInfoForDatasource(datasource)
+      return info.views?.map(v => v.name) || []
+    } catch (error) {
+      // Views not supported for this datasource
+      return []
+    }
+  }
+
+  async createViewQuery(datasource: Datasource, viewName: string | undefined) {
+    if (!viewName) return
+    try {
+      const info = await API.fetchViewInfoForDatasource(datasource)
+      if (info.error) {
+        console.error("View definitions error:", info.error)
+        return
+      }
+      const view = info.views?.find(v => v.name === viewName)
+      const sql =
+        "SELECT * FROM " + view?.name + " OFFSET {{ offset }} LIMIT {{ limit }}"
```

Comment:
Use the mandatory viewName from the signature, and also add a newline `\nOFFSET` and `\nLIMIT` just to make the query text have a nice format when imported